v4l2/v4l2_codecbase.cpp

```diff
@@ -391,7 +391,7 @@ int32_t V4l2CodecBase::ioctl(int command, void* arg)
                 ERROR("unknown request buffer type: %d", reqbufs->type);
                 break;
             }
-            ASSERT(port == INPUT || reqbufs->memory == m_memoryMode[port]);
+            // ASSERT(port == INPUT || reqbufs->memory == m_memoryMode[port]);
```

Comment:
VIDIOC_REQBUFS init memory type of input/output buffer.
1.  chromeos assumed input/output buffer memory type is MMAP, so we init m_memoryModep[port] to MMAP and ASSERT() here
2. next, we add USRPTR memory type for input buffer, it means 'port == INPUT' needn't assert memory type (but accept memmory type set from client).
3. now we add ANDROID_BUFFER_HANDLE memory type for output buffer, then 'port == OUTPUT' needn't ASSERT memory type either.